airflow/jobs/scheduler_job.py

```diff
@@ -1593,7 +1592,9 @@ def _create_dag_runs(self, dag_models: Iterable[DagModel], session: Session) ->
             # we need to run self._update_dag_next_dagruns if the Dag Run already exists or if we
             # create a new one. This is so that in the next Scheduling loop we try to create new runs
             # instead of falling in a loop of Integrity Error.
-            if (dag.dag_id, dag_model.next_dagrun) not in active_dagruns:
+            if (dag.dag_id, dag_model.next_dagrun) not in active_dagruns and len(
+                active_dagruns
+            ) < dag.max_active_runs:
```

Comment:
Two things:

1) This variable is badly named, and the "active" prefix is incorrect -- it turns out it is just _all_ dagruns, not just active/running ones
2) len(active_dagruns) includes all dags, not just the current `dag.dag_id`

so this check needs changing.

Reply:
I have corrected it. Surprised why it was working previously. Thanks